sensor/InternalScanSensor.js

```diff
@@ -40,22 +40,46 @@ const am2 = new AM2();
 const sem = require('../sensor/SensorEventManager.js').getInstance();
 const featureName = "internal_scan";
 
+const extensionManager = require('./ExtensionManager.js')
+
 class InternalScanSensor extends Sensor {
   constructor() {
     super();
   }
 
-  async run() {
+  async apiRun() {
+
     this.running = false;
     this.supportPorts = ["tcp_23", "tcp_80", "tcp_21", "tcp_3306", "tcp_6379"]; // default support: telnet http ftp mysql redis
     if (platform.getName() === 'gold') {
       this.supportPorts.push("tcp_22"); // gold: ssh
     }
-    //this.supportPorts = ["tcp_23"];
-    sem.once("DeviceServiceScanComplete", async (event) => {
-      await this.checkAndRunOnce();
+
+    extensionManager.onGet("getScanStatus", async (msg, data) => {
+      return this.scanStatus;
+    });
+
+    extensionManager.onCmd("killScanSession", async (msg, data) => {
+      return this.stop();
+    });
+
+    extensionManager.onCmd("startScanSession", async (msg, data) => {
+      if (!this.running) {
+        await this.runOnce();
+        return {"msg":"start scan"} 
+      } else {
+        return {"msg":"previous scan is running"} 
+      }
     });
 
+  }
+
+  async stop() {
+    let cmd = "sudo pkill -9 nmap";
```

Comment:
This may kill other nmap processes that are initiated by other sensors. Better to record the PID of the currently running nmap process in this sensor and kill the specific PID